### PR TITLE
Create KeyframeAnimation for empty keyframe rules

### LINKED ISSUE #15257: Create css animation object even if corresponding keyframe is empty
From spec[1]
> Any animation for which a valid keyframe rule is defined will run and generate events; this includes animations with empty keyframe rules.

[1] https://drafts.csswg.org/css-animations-1/#events

Also a web animations API, getFrames[2] expects an empty keyframe for such cases.

[2] https://w3c.github.io/web-animations/#dom-keyframeeffectreadonly-getkeyframes

### PR DESCRIPTION
**Pull request by hiikezoe**
We should create KeyframeAnimation even if css keyframe rule is empty because we should fire css animation events for such CSS animations.
Also we should ignore properties that are annotated with '!important'.  From the spec[1].
 using !important on them is invalid and will cause the property to be ignored

[1] https://drafts.csswg.org/css-animations-1/#keyframes
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15257 

- [X] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15299)

### CODE REVIEW

File: tests/unit/style/keyframes.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+#[test]
+fn test_no_property_in_keyframe() {
+    let keyframes = vec![
+        Arc::new(RwLock::new(Keyframe {
+            selector: KeyframeSelector::new_for_unit_testing(vec![KeyframePercentage::new(1.)]),
+            block: Arc::new(RwLock::new(PropertyDeclarationBlock {
+                declarations: vec![
+                ],
```
Comment (pcwalton) [inline tests/unit/style/keyframes.rs]: nit: weird indentation here

File: components/style/keyframes.rs [hunk 2]
```diff
@@ -254,7 +254,11 @@ fn get_animated_properties(keyframe: &Keyframe) -> Vec<TransitionProperty> {
     let mut ret = vec![];
     // NB: declarations are already deduplicated, so we don't have to check for
     // it here.
-    for &(ref declaration, _) in keyframe.block.read().declarations.iter() {
+    for &(ref declaration, importance) in keyframe.block.read().declarations.iter() {
+        if importance.important() {
```
Comment (pcwalton) [inline components/style/keyframes.rs]: Add a comment here specifying that the spec requires this

Comment (emilio) [inline components/style/keyframes.rs]: This isn't needed at all, we drop them during parsing so you should be able to `assert!(!importance.important())` here.

Comment (hiikezoe) [inline components/style/keyframes.rs]: Thanks! I did not notice it. You know anything about servo's styling!

**Comment by emilio** [comment]
Looks good, thanks!

@bors-servo r=emilio,pcwalton

**Pull request merged**